pylint/checkers/strings.py

```diff
@@ -82,7 +82,7 @@
 DOUBLE_QUOTED_REGEX = re.compile('(%s)?"""' % "|".join(_PREFIXES))
 QUOTE_DELIMITER_REGEX = re.compile("(%s)?(\"|')" % "|".join(_PREFIXES), re.DOTALL)
 
-MSGS = {
+MSGS = {  # pylint: disable=consider-using-namedtuple
```

Comment:
Isn't this an example that this will be prone to false positives ? We had nothing to fix in the codebase (And I think there is dict that could be named tuple in the MessageStore/MessageIdStore), but we have a false positives on our checkers.

Reply:
Technically I wouldn't consider that a false-positive. A NamedTuple would be better here but that requires a bigger refactoring which is why I've added `pylint: disable` for now.

You are right that this checker might result in a few false-positives, but ultimately I believe it is never the less worth it. I have seen many occurrences of something like this code here (the comment wasn't added by me):
```py
# Sensor types: Name, device_class, event
SENSOR_TYPES = {
    "button": ["Button", DEVICE_CLASS_OCCUPANCY, "device:sensor:button"],
    "motion": ["Motion", DEVICE_CLASS_MOTION, "device:sensor:motion"],
}
```

Reply:
That's fair. But maybe we should not activate it by default ? What do you think ? (As an aside refactoring the Checker and or Pylinter class is a humongous task, do you have an idea how you would like to do it ideally ?)

Reply:
Had to think about your comment a bit.

> But maybe we should not activate it by default ? What do you think ?

Though about this as well initially. If the checker would result in too many true false-positives, making it optional would be the better solution. However the drawback is that it will be almost useless as not many will try it.

I think of it this way: Aside from a few instances where module level constants are modified (which is bad code by itself), the checker works as intended. Additionally it's only a `refactoring` suggestion, not a warning. And in the end it can always be completely disabled by the user. My only hope is they first discover that the suggestions in deed make sense and thus keep it activated.

> As an aside refactoring the Checker and or Pylinter class is a humongous task, do you have an idea how you would like to do it ideally ?

When I added the `pylint: disable` statements, I did not. As it turns out, it's way easier than expected 😅 Especially since the message definitions are tuples already.

Reply:
>  As it turns out, it's way easier than expected 😅 Especially since the message definitions are tuples already.

So this ``MsgDef`` class is not a breaking change for checkers using it ? Could we do it in another MR and put it in ``pylint.messages`` ? There is probably some rationalization to do between this class and ``MessageDefinition`` / ``MessageId`` as there is a concept of msgid (C0111), symbol (missing-docstring), message "Missing docstring in class or function %s), description (Raise a warning when a docstring is missing blablabla), and message definition (a class very similar to the one you created) already. Maybe there is some performance to gain with this too on top of the clarity

Reply:
👍🏻 I've removed the related commits here and will do them in another MR